template/base/src/env/client.mjs

```diff
@@ -14,7 +14,7 @@ export const formatErrors = (
     })
     .filter(Boolean);
 
-if (!_clientEnv.success) {
+if (_clientEnv.success === false) {
```

Comment:
How come this would help?

Reply:
Honestly I'm not sure @juliusmarminge. The bang operator can be strange sometimes, but in this case `!_clientEnv.success` _should_ behave the same way as `_clientEnv.success === false`. However, it clearly exhibits unexpected behavior given that at least two of us have gotten this error now, which the change I am proposing would fix. 

If you look at the types for Zod, you can see that `success` is a required property, so we are not changing the logic by switching from `!` to `=`. 

https://github.com/colinhacks/zod/blob/master/src/types.ts#L128

Reply:
Yea I guess we could try. Would you like to submit this change to the next-branch as well?